utils/src/main/java/software/amazon/awssdk/utils/async/FilteringSubscriber.java

```diff
@@ -40,11 +40,17 @@ public void onSubscribe(Subscription subscription) {
 
     @Override
     public void onNext(T t) {
-        if (predicate.test(t)) {
-            subscriber.onNext(t);
-        } else {
-            // Consumed a demand but didn't deliver. Request other to make up for it
-            subscription.request(1);
+        try {
+            if (predicate.test(t)) {
+                subscriber.onNext(t);
+            } else {
+                // Consumed a demand but didn't deliver. Request other to make up for it
+                subscription.request(1);
+            }
+        } catch (RuntimeException e) {
+            // Handle the predicate throwing an exception
+            subscription.cancel();
```

Comment:
Query just for my curiosity : Earlier these exceptions were thrown higher in stack and now we are catching it , do we need to worry about any backward compatibility issue were Users have expecting this exceptions to be caught directly.

Reply:
It's against "the rules" to throw out of an onNext (except for null parameters), so we're out of compliance already here. It's possible someone is relying on the old behavior: https://xkcd.com/1172/

Reply:
Thanks Matt.